src/gravity.js

```diff
@@ -0,0 +1,508 @@
+
+  } else if (!wasStandingOnSomething) {
+
+    // if none of the other options are valid, and they are falling, then make em fall!
+    reallyMovePlayer(player, gravityDir);
+    lastMoveMapped = 'down';
+
+  } else {
+    // TODO: DIS WHAT HAPPENS WHEN YR STANDING ON THE EDGE!
+  }
+}
+
+// DANGER!
+
+// BRITTLE CODE
+inject(/player\(\)\.x -= 1;/, "movePlayerWithGravity('left', 'x', -1);");
+inject(/player\(\)\.x \+= 1;/, "movePlayerWithGravity('right', 'x', 1);");
+inject(/player\(\)\.y -= 1;/, "movePlayerWithGravity('up', 'y', -1);");
+inject(/player\(\)\.y \+= 1;/, "movePlayerWithGravity('down', 'y', 1);");
+
+// EXTREMELY BRITTLE CODE
+inject(/var ext = getExit\( player\(\)\.room, player\(\)\.x, player\(\)\.y \);/, "else { advanceGravity(); didPlayerMoveThisFrame = true; } var ext = getExit( player().room, player().x, player().y ); ");
+
+// END OF DANGER!
+
+/* HELPERS
+*/
+
+function mapDir(dir) {
+  return gravityMap[gravityDir][dir];
+}
+
+function invertPhysics(newDown) {
+  if (isValidDir(newDown)) {
+    gravityDir = newDown;
+  } else {
+    console.warn('gravity: invertGravity failed, expected "up|down|left|right", but got ', newDown);
+  }
+}
+
+function isValidDir(dir) {
+  return dirs.indexOf(dir) !== -1;
+}
+
+function capitalize(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+function isSolid(dir, x, y) {
+  var capDir = capitalize(dir);
+  var wallCheck = `isWall${capDir}`;
+  var spriteCheck = `getSprite${capDir}`;
+  var isSpriteThere =  bitsy[spriteCheck]()
+  var edgeMap = {
+    up: { coord: y, value: 0 },
+    down: { coord: y, value: 15 },
+    left: { coord: x, value: 0 },
+    right: { coord: x, value: 15 }
+  }
+  var edgeToCheck = edgeMap[dir]
+  var isWallThere = bitsy[wallCheck]() || (edgeToCheck.coord == edgeToCheck.value)
+
+  return isWallThere || isSpriteThere;
+}
+
+function isTileClimbable(x, y) {
+  var tile = bitsy.tile[bitsy.getTile(x, y)];
+  return tile && tile.name && hackOptions.isClimbable(tile)
+}
+
+function isOnStandableTile(player) {
+  if (fallCounter > 1) return;
+  var coords = [player.x, player.y]
+  var offset = offsets[gravityDir]; // like [0, -1] for y -= 1
+  coords[0] += offset[0]
+  coords[1] += offset[1]
+  var tile = bitsy.tile[bitsy.getTile(coords[0], coords[1])];
+  return tile && tile.name && hackOptions.isStandable(tile)
+}
+
+function canMoveHorizontallyWhileFalling() {
+  var withinMaxRatio = (horizontalFallingMoves / fallCounter) <= hackOptions.maxHorizontalFallingRatio;
+
+  // if fallCounter is 0 (start of fall/jump) or player last moved down and is within ratio
+  return !fallCounter || (lastMoveMapped === 'down' && withinMaxRatio);
+}
+
+function reallyMovePlayer(player, dir) {
+
+  if (isSolid(dir, player.x, player.y)) {
+    // can't move into solid thing...so...chill?
+    // should maybe trigger sprites here?
+    return;
+  }
+
+  // why doesn't isSolid catch the out of bounds? stuff? isWall should as well? weird...
```

Comment:
im pretty sure bitsy just checks the edges like you're doing here; would be best to reference `bitsy.mapsize` instead of the hard-coded `15`s though